sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/console.scala

```diff
@@ -49,7 +49,7 @@ class ConsoleSink(options: Map[String, String]) extends Sink with Logging {
     println("-------------------------------------------")
     // scalastyle:off println
     data.sparkSession.createDataFrame(
-      data.sparkSession.sparkContext.parallelize(data.collect()), data.schema)
```

Comment:
@marmbrus Michael, that's unfortunate. The OOM risk might be common to any source that can deliver a high volume of data at once (file source in my test case, but I would expect that a loaded kafka topic read from `earliest` will behave in the same way).